test/Dialect/FIRRTL/errors.mlir

```diff
@@ -871,3 +871,115 @@ firrtl.circuit "Foo"   {
     %bar_a, %bar_b, %bar_c = firrtl.instance bar sym [<@w3,1,public>,<@w3,2,private>,<@syh2,0,public>] @Bar(in a: !firrtl.uint<1> [{one}], out b: !firrtl.bundle<baz: uint<1>, qux: uint<1>> [{circt.fieldID = 1 : i32, two}], out c: !firrtl.uint<1> [{four}])
   }
 }
+
+// -----
+// Check single usage of reference ports
+firrtl.circuit "Foo" {
+  firrtl.module @Bar(in %_a: !firrtl.ref<uint<1>>) {
+    %a = firrtl.wire : !firrtl.uint<1>
+    firrtl.ref.recv %a, %_a : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Foo(in %fa: !firrtl.ref<uint<1>>) {
+    %bar_a = firrtl.instance bar @Bar(in _a: !firrtl.ref<uint<1>>)
+
+    %zero = firrtl.constant 0 : !firrtl.uint<1>
+    // expected-error @+1 {{reference port operand cannot be reused by any other op}}
+    %0 = firrtl.ref.resolve %bar_a : !firrtl.ref<uint<1>>
```

Comment:
Stated previously: this is quite restrictive. I don't really see any problem with this circuit. That said, I think this is a reasonable place to start.

Reply:
So, this ensures that there is only one `ref.resolve` for the corresponding `ref.recv/send`. If we want to allow multiple XMRs from the same reference ports then we can remove the restriction.